src/components/memorygame/gameHelpers.jsx

```diff
@@ -0,0 +1,34 @@
+// manipulates isFlipped and isMatched property of card a state
+// property example : "isMatched"
+// cardStates is an array of objects(states)
+// see src/memorygame/MemoryGameBoard/index.jsx to for cardStates example and use of this function
+// targetCard is an element of cardStates array.
+// changeCardStatePropertyToOpposite is called when selectedCards state change.
```

Comment:
`invertCardStateForKey is called ...`

Reply:
majd has some really good points here in general tarkan, which for both of you, the general principle will be a good takeaway. it's good that both of you are writing more comments and developing an intuition for whats readable. the next step is also to write "maintainable" comments

even a small change requested can cause your comments be outdated. and we (we as in we programmers) dont have time to remember comment dependencies (even in a typed language like typescript, this is not "typed" properly in comments, so your code will still work even if comments are outdated)

the lesson / goal here is to write your comments to be as futureproof as possible (its not completely possible and you dont want to be vague but its a goal to aim for). for example, avoiding line numbers. or, when you're naming props, you can put the comment above the prop for example without mentioniing its name (in JS theres not a good way to do this). thus even if you rename the prop the comment is still above the prop

Reply:
By saying `you can put the comment above the prop for example without mentioniing its name` are you referring to something like below? @louisrli 
```
function(
  // does bla bla
  param1,
  // does different bla
  param2
)
```